### ISSUE TITLE
Recommend a decentralized and full P2P ID provider, and enable it on official sites

### ISSUE THREAD

**Issue by DaniellMesquita** [HelloZeroNet/ZeroNet#1646]

**Comment by DaniellMesquita** [comment]
@krixano

**Comment by krixano** [comment]
I did make a PR on ZeroTalk. One of the things that's really needed before very wide acceptance of KxoId is that I need a better computer because the one that I use right now freezes after a few days.

**Comment by DaniellMesquita** [comment]
Instead of a better computer, more trusted nodes.

**Comment by HelloZeroNet** [comment]
https://github.com/HelloZeroNet/ZeroNet/issues/1258

The POW based ZeroID providers are still planned and would be relatively easy to add. I started to generate a "1ZeroiD[0-9]", but it never finished, I will fire it up again.

**Comment by DaniellMesquita** [comment]
@shortcutme 
What about unique names without relying on server?

**Comment by HelloZeroNet** [comment]
Not sure if it's possible

**Comment by DaniellMesquita** [comment]
If you use trusted nodes like KxoID

**Comment by Thunder33345** [comment]
well it still relies in something to keep everything in order..

a truly serverless nodeless system can:
utilize bitcoin cryptography itself, ever notice what's the odds of 2 bitcoin address begin the same?
we can just have user's names as bitcoin address

of course we lose the ability to pick handlers but it's the price we pay
Also see: [Zooko's Triangle](https://en.wikipedia.org/wiki/Zooko's_triangle)
![](https://upload.wikimedia.org/wikipedia/commons/thumb/5/5c/Zooko%27s_Triangle.svg/450px-Zooko%27s_Triangle.svg.png)
we can never really  pick all of it at once

but for truly self maintained ID(id that can be generated without connection to any servers or approval by any server yet maintains uniqueness) will take more work and probably should be in another issue

**Comment by rllola** [comment]
@Thunder33345 
Namecoin isn't suppose to be the first solution to Zooko Triangle ?
```
Namecoin was also the first solution to Zooko’s Triangle, the long-standing problem of producing a naming system that is simultaneously secure, decentralized, and human-meaningful.
```
https://namecoin.org/

Why the need for something else ?

**Comment by Thunder33345** [comment]
i mean i am assuming our endgoal is to have a trustless and decentralized system with always unique username
i think unique enforced by crypto is always better then begin enforced by trusted authority
namecoin works by all actions costing money, double edged sword there

>Why the need for something else ?
what you meant by? we could also use namecoin for usernames but who want to pay every time for a username
but the idea of a system that dont rely on trust is nice

but using raw bitcoin address as username would be somewhat customisable yet 100% unique and decentralized yet trustless

**Comment by rllola** [comment]
> i mean i am assuming our endgoal is to have a trustless and decentralized system with always unique username

No my bad I though we were talking about DNS (I still had my head on an other issue). Namecoin doesn't fill the unique id purpose.

You are right current system is centralized and it is not great.
We have 3 solutions : 
 - Namecoin but like you said you need to pay and you need to buy namecoins... It is not super userfriendly... Yet it happened only once.
 - Proof-of-Work ID but it is not user friendly really and you don't have the advantage of human-meaningful name.
 - Leave it centralized (or semi-centralized we could have several servers and validators in case the server is going down but it would still be possible for validators to come together and cheat).

I say pick the less worst solution :smile:

**Comment by Thunder33345** [comment]
re POW:

we can have something like POWless but use bitcoin crypto so it's just impossible or financially unviable to create the same bitcoin address but free to create an user
these who want readable names can use POW to generate something somewhat readable, the more determined character means more POW cost
i will be honest dragging whatever characters after the prefix dosent look pretty either but it's some sort of possible compromise

we can also somehow create our own crypto(sounds bad but lets just assume it on paper)
which would have something like a 2 part system there would be "display name" which is "easier to regenerate to our liking" and a "unique id" part of the name which is longer and harder to duplicate(could also derive unique avatar out of said unique ID to make it impossible to impersonate)

drawbacks:
major backward compatibility breaking
how to even make it compatible with the current system is yet to be figured
how to actually make it a reality is yet to be figured
inhouse developed protocols might be prone to exploits thus need even more testing before using

which also reminds me of some self singing cert providers why not just use them, they already can customise the username display, and the auth address is always unique, we can derive an identifier based off the auth address to each user so they cant be impersonated and afterthought: there's not much differences other then the fact of the ladder already existed...

now by that said
centralized will have to make do with for now
but we can do better, and we should do better in the future

it's a step from finally stop needing clearnet request to get an ID so that's an improvement i will take,
it still needs trusted bodies but it nolonger need clearnet
